src/coreclr/build-runtime.cmd

```diff
@@ -373,9 +379,24 @@ if %__BuildNative% EQU 1 (
         set __ExtraCmakeArgs="-DCMAKE_BUILD_TYPE=!__BuildType!"
     )
 
-    set __ExtraCmakeArgs=!__ExtraCmakeArgs! "-DCLR_CMAKE_TARGET_ARCH=%__TargetArch%" "-DCLR_CMAKE_TARGET_OS=%__TargetOS%" "-DCLI_CMAKE_FALLBACK_OS=%__HostFallbackOS%" "-DCLR_CMAKE_PGO_INSTRUMENT=%__PgoInstrument%" "-DCLR_CMAKE_OPTDATA_PATH=%__PgoOptDataPath%" "-DCLR_CMAKE_PGO_OPTIMIZE=%__PgoOptimize%" %__CMakeArgs%
-    echo Calling "%__RepoRootDir%\eng\native\gen-buildsys.cmd" "%__ProjectDir%" "%__IntermediatesDir%" %__VSVersion% %__HostArch% %__TargetOS% !__ExtraCmakeArgs!
-    call "%__RepoRootDir%\eng\native\gen-buildsys.cmd" "%__ProjectDir%" "%__IntermediatesDir%" %__VSVersion% %__HostArch% %__TargetOS% !__ExtraCmakeArgs!
+    if /i not "%__HostOS%" == "%__TargetOS%" (
+        if /i "%__HostOS%" == "" (
+            set __ExtraCmakeArgs=!__ExtraCmakeArgs! "-DCLR_CMAKE_TARGET_ARCH=%__TargetArch%" "-DCLR_CMAKE_TARGET_OS=%__TargetOS%"
+        )
+    )
+
+    set __ExtraCmakeArgs=!__ExtraCmakeArgs! "-DCLI_CMAKE_FALLBACK_OS=%__HostFallbackOS%" "-DCLR_CMAKE_PGO_INSTRUMENT=%__PgoInstrument%" "-DCLR_CMAKE_OPTDATA_PATH=%__PgoOptDataPath%" "-DCLR_CMAKE_PGO_OPTIMIZE=%__PgoOptimize%"
+
+    if /i "%__TargetOS%" == "android" (
+        if not "%__HostOS%" == "" (
+            set "__TargetOS=!__HostOS!"
+        )
+    )
```

Comment:
Instead of special-casing Android here, could we special case it in `runtime.proj` instead? Mainly to help reduce the amount of batch we have to deal with.

Reply:
I tried, but all roads lead to some kind of bash change. If you special case in runtime.proj around `-os`, the artifacts directory becomes top level instead of <arch> below the original target os. To get around that requires... more batch.

I think we keep it where it is and then at some point, just convert the whole batch file into powershell / incorporate your cmake arcade thing.